PR for review in parallel of  UX library development

alexandercyu opened the pull request:
Added from Figma design 
- Added colours
- Added fonts
- Added buttons (hover states and pressed states implemented for some buttons only)


**Update 2024-03-20**
- Added technology card as a partial
- Added content section as a partial
- Added technology section to index as a test
- Added funder logos (a few still need to be added)
- Added footer as a partial
- Added footer to index as a test

**Update 2024-03-22**
- Added Users section (student, researcher, hobbyist section)
- minimum and maximum width/height for footer logos set

**Update 2024-03-25**
- Added FAQ section
- Added CSS for FAQ section
- Changed spacing between elements such as titles, leading text, buttons, content sections, and paragraphs/text sections so they are in line with the design
- Centered text content areas but not image content areas, as shown in the design

**Update 2024-03-27**
- Added Extras section
- Extras includes Feedback, Useful Links, and social media
- Fixed colour changing for Get Started menu

**Update 2024-04-02**
- Put all colours in root class in css
- first draft of nodes page done


**Update 2024-02-03**
- Update menu, banner, and quickstart menu with styles from figma design
- Added positioning so top menu is overlayed on banner video
- Added menu.js to change menu text colours depending on which page it is
- Moved used styles from daccs-css.css into ux-library.css
- Removed alert bar from top of page


**Update 2024-04-04**
- Updated menu so spaces between menu items are 80px, font size is h6
- Updated quickstart menu to remove solid border, add elevation (box shadow)
- Corrected h6 font size in css to 20px
- Added introduction section (About Marble) to home page

**ToDo**

- Nodes page (desktop version)

mishaschwartz commented on static/css/marble.css:
```diff
@@ -0,0 +1,399 @@
+    background-color: #FFFFFF;
+}
+
+.surface-alt{
```
Is this meant to be empty?

alexandercyu replied:
I had a conversation with Shruti last night and this morning about the alternative design for the button.  It uses a different colour scheme but the colours are not defined as a primary alternative or surface alternative.  Shruti says the design can only have one primary and one surface colour defined, but an alternative can be defined in the CSS if needed.  I left this blank for the time being.

mishaschwartz replied:
Ok, that makes sense. Any time you do something like this though please leave a comment in the code explaining why.

alexandercyu replied:
Since last update the surface-alt class is no longer needed.  It is deleted.

mishaschwartz commented on templates/partials/button.html:
```diff
@@ -0,0 +1,48 @@
+    {% set button_background_colour = "bg-button-primary" %}
+    {% set button_font = "button-text" %}
+    {% set button_font_colour = "text-on-primary" %}
+{% endif %}
```
if these if statements are not meant to overlap, use if/elif

alexandercyu replied:
Changed to use elif.

mishaschwartz commented on templates/site/test-button.html:
```diff
@@ -0,0 +1,46 @@
+
+{% block content %}
+    
+    {% with med_action_button = true%}
```
why not have a `button_type` variable instead of setting variables to true? Wouldn't that be more readable?

mishaschwartz commented on templates/site/test-button.html:
```diff
@@ -0,0 +1,46 @@
+{% block content %}
+    
+    {% with med_action_button = true%}
+        {% set button_disabled = "" %}
```
Make this the default so it doesn't have to be explicitly stated every time for non-disabled buttons.

alexandercyu replied:
Disabled status made default.

mishaschwartz commented on templates/site/test-button.html:
```diff
@@ -0,0 +1,46 @@
+    {% endwith %}
+
+    {% with med_action_button_alt = true%}
+        {% set button_disabled = "disabled" %}
```
If it's disabled then make this true and apply the appropriate classes in the partial. As it is right now, I could add any text here and it would be used as a new class.

alexandercyu replied:
Changed disabled toggle to use true.

alexandercyu commented on templates/site/test-button.html:
Added a comment as a reminder to remove before merging.

alexandercyu commented on templates/partials/button.html:
```diff
@@ -0,0 +1,48 @@
+
+    {% if single_action_button %}
+        <button class="{{ button_type }} {{ button_background_colour }} {{ button_font }} {{ button_font_colour }} {{ button_disabled }}" type="button">
+            <svg width="{{ svg_width }}" height="{{ svg_height }}" viewBox="0 0 {{ viewbox_width }} {{ viewbox_height }}" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Shruti's design used an SVG.  I thought she had a reason for using it.

mishaschwartz replied:
@ShrutiK100 is there a reason to draw our own icons instead of using a library?

ShrutiK100 replied:
Also I have not added any icons in my design library yet

alexandercyu replied:
We use fontawesome   for our icons.

https://fontawesome.com/start

alexandercyu commented:
**Update 2024-03-20**
- Added technology card as a partial
- Added content section as a partial
- Added technology section to index as a test
- Added funder logos (a few still need to be added)
- Added footer as a partial
- Added footer to index as a test

alexandercyu commented:
**Update 2024-03-22**
- Added Users section (student, researcher, hobbyist section)
- minimum and maximum width/height for footer logos set

**ToDo**
- Menu for Users section needs menu item focus fixed so the highlighted colour will stay when the menu item is clicked

alexandercyu commented:
**Update 2024-03-25**
- Added FAQ section
- Added CSS for FAQ section
- Changed spacing between elements such as titles, leading text, buttons, content sections, and paragraphs/text sections so they are in line with the design
- Centered text content areas but not image content areas

**ToDo**
- Social media links
- Contact section
- Other links section
- Nodes page (desktop version)
- Mobile layout of home page

mishaschwartz commented on static/css/ux-library.css:
```diff
@@ -0,0 +1,584 @@
+}
+
+h1{
+    font-family: Arial, Helvetica, sans-serif;
```
In the design document, the primary font for h1 is Helvetica not Arial

alexandercyu replied:
Switched fonts around so Helvetica is first

mishaschwartz commented on templates/partials/button.html:
I think we need to rethink this file. 
I would really like to be able to manage most of this logic in the css. That means that we should be avoiding svg buttons if possible as long as there is an appropriate icon. 

Let's discuss this offline

mishaschwartz replied:
We don't have to resolve this here but please make an issue or something so that we can come back to this file later on.

mishaschwartz commented on templates/partials/content-section.html:
```diff
@@ -0,0 +1,55 @@
+<div class="container-fluid content-section">
+    <div class="row mt-5">
+        {% if template %}
```
All of these if/elif blocks should be their own partials

alexandercyu replied:
Moved all these to the index to be used as partials.  
Deleted content-section.html.

mishaschwartz commented on templates/partials/feedback.html:
```diff
@@ -0,0 +1,18 @@
+<div>
+    <h3>{{ feedback_title }}</h3>
```
Do we need this as a variable? Is the title ever going to be anything other than "Feedback"?

alexandercyu replied:
Good point.  Just had the variable for consistency with the others I guess.

mishaschwartz commented on templates/partials/feedback.html:
The content here has changed (see figma)

alexandercyu replied:
Removed unneeded feedback entry.

mishaschwartz commented on templates/partials/footer.html:
Can we align the logos in the centre of the screen and add more padding between the logos please

alexandercyu replied:
Centered logos and added 40px padding between logos.

mishaschwartz commented on static/css/ux-library.css:
```diff
@@ -0,0 +1,584 @@
+}
+
+.funder-logo{
+    min-width: 10%;
```
min-width should be a fixed value. 10% of a tiny screen is potentially unreadable

alexandercyu replied:
Changed min width to 100px.

mishaschwartz commented on templates/partials/footer.html:
```diff
@@ -1,13 +1,32 @@
+
+ <div class="container-fluid footer">
+    <div class="row d-flex flex-sm-column flex-md-row">
+        <div class="align-items-center">
```
this class should be on the parent div (it isn't doing anything at the moment)

alexandercyu replied:
Changed to the parent and changed to justify-content-centre.

mishaschwartz commented on templates/partials/social_media.html:
```diff
@@ -0,0 +1,24 @@
+    <h3>{{ social_media_title }}</h3>
+
+    {% set link_icon_array = ["images/logos/social-media/icon_Linkedin.svg",
+    "images/logos/social-media/facebook-v1.svg",
```
I don't think we have a facebook account

alexandercyu replied:
Removed the icon

mishaschwartz commented on templates/partials/social_media.html:
```diff
@@ -0,0 +1,24 @@
+
+    <div>
+        {% for array_item in range(array_length) %}
+            <a href="{{ link_url_array[array_item] }}"><img class="social-media-icon" src="{{ link_icon_array[array_item] }}"></a>
```
Icons stack up too close together vertically when screen is small

alexandercyu replied:
Added 15px margin to top of icon.

mishaschwartz commented on templates/partials/useful-links.html:
```diff
@@ -0,0 +1,14 @@
+<div>
+        <h3>{{ useful_links_title }}</h3>
+
+        {% set link_text_array = ["Get started as an admin", "Deploy a node", "Marble Project Website", "Node Registry", "Python Client"] %}
```
Isn't the marble project website this website??

alexandercyu replied:
It's on the design right now.  I guess the final list will be decided later.

mishaschwartz commented on templates/site/design-component-preview.html:
```diff
@@ -0,0 +1,126 @@
+{# TODO: This file is for testing components.  Remove this before merging. #}
```
If this is ready to be merged, please remove

alexandercyu replied:
Perhaps keep this as a preview page?  Move it to temp or something so the components can be viewed.

mishaschwartz replied:
This is useful for developers but shouldn't be visible on the live site. Please remove it from the `site` directory before this is merged.

mishaschwartz commented on templates/site/js/usage.js:
```diff
@@ -0,0 +1,39 @@
+displayArea.innerHTML = studentUser.innerHTML;
+
+studentButton.onclick = function () {
+    studentButton.style.color = "#304FFE";
```
use classes, don't assign style directly

alexandercyu replied:
Changed to assign classes.

mishaschwartz commented on templates/site/test-node-card.html:
If this is for testing, please remove before merging

alexandercyu replied:
This is for the nodes page I'm working on.

alexandercyu commented on static/css/daccs-css.css:
New styles are in the ux-library.css.. Old stuff is in the daccs-css.css and is currently used for the banner and menu, that haven't been updated yet.  The daccs-css.css will be deleted and any unduplicated css will be copied over.

alexandercyu commented on static/css/ux-library.css:
Made a main font variable for 'Helvetica Neue', Helvetica, Arial,  sans-serif; 
and secondary font variable for Helvetica, Arial,  sans-serif;

mishaschwartz commented on build.py:
The changes in this file shouldn't be shown here. Please make sure that the `ux-feature-branch` is up to date with the `main` branch

mishaschwartz commented on static/css/ux-library.css:
```diff
@@ -0,0 +1,580 @@
+}
+
+.button-sm:hover, .button-med:hover{
+    background-color: #000000;
```
use variable

(this comment applies to all other hardcoded styles that are still left in this file)

alexandercyu replied:
Converted remaining colours to variables

mishaschwartz commented on templates/layouts/main.html:
```diff
@@ -9,14 +9,14 @@
 </head>
 <body>
 {% include "partials/menu.html" %}
-{% include "partials/alert.html" %}
```
If it's not used anymore remove the file as well

mishaschwartz commented on templates/site/js/menu.js:
```diff
@@ -0,0 +1,54 @@
+    var tutorialsButton = document.getElementById('tutorialsMenuButton');
+    var communityButton = document.getElementById('communityMenuButton');
+    var aboutButton = document.getElementById('aboutMenuButton');
+    console.log("menu function running");
```
remove please

alexandercyu replied:
Removed

mishaschwartz commented on templates/site/js/menu.js:
```diff
@@ -0,0 +1,54 @@
+    var currentPath = window.location.pathname;
+
+    if (currentPath.includes("index.html") && currentURL.includes("index.html#technology") == false) {
+        console.log("home clicked");
```
remove please

alexandercyu replied:
This is removed.

mishaschwartz commented on templates/site/node.html:
```diff
@@ -1,62 +1,66 @@
+        {% set section_title = "Start a Project" %}
+        {% set section_content = "Select most suitable node or server to start a project" %}
+
+        {% include "partials/title-and-text.html" %}
```
partial doesn't exist

alexandercyu replied:
Partial renamed to general-content.html

mishaschwartz commented on templates/site/node.html:
```diff
@@ -1,62 +1,66 @@
+        <div class="row">
+            <div id="nodeMenu"></div>
+        </div>
+<!--
```
remove

alexandercyu replied:
Commented parts removed, but page still in development.

mishaschwartz commented on templates/site/js/node.js:
```diff
@@ -112,58 +123,115 @@ document.addEventListener("DOMContentLoaded", function () {
\ No newline at end of file
+})
+
+function getNode(){
```
empty?

alexandercyu replied:
Pushed non-empty function

mishaschwartz commented on templates/site/js/node.js:
```diff
@@ -112,58 +123,115 @@ document.addEventListener("DOMContentLoaded", function () {
+                    }
+
+
+                    //else{
```
remove

alexandercyu replied:
Removed

mishaschwartz reviewed (changes requested):
I can't build this right now because there's a missing partial. I left a few other comments but please add that partial in and then I'll continue the review

mishaschwartz commented on templates/partials/menu.html:
```diff
@@ -14,26 +14,27 @@
                 <div class="col-10">
```
The midline of the menu items should be aligned with the centre of the logo

mishaschwartz commented on templates/partials/banner.html:
```diff
@@ -1,15 +1,14 @@
-<section id="hero" class="page-section first-section text-white">
+<section id="hero" class="banner-section first-section text-white">
     <div id="banner-image" class="d-block d-lg-none bg opacity-30 bg-position-center-center bg-attachment-fixed"></div>
```
Banner image is no longer visible?

alexandercyu replied:
It's a side-effect from my work on the nodes page.  It's not showing on there and both pages use the same template.

alexandercyu replied:
Banner image is fixed.

mishaschwartz commented on templates/partials/introduction.html:
```diff
@@ -0,0 +1,13 @@
+            <h3>{{ title }}</h3>
+            <div>{{ content|safe }}</div>
+        </div>
+         {% with button_type = "med_action_button"%}
```
This button looks very different on Figma: there's more padding around the text, the text is vertically centred, the text is smaller, 

Can you please double check the css

alexandercyu replied:
I think the issue here is the template in the design for a medium button sets the width at 108px but the design for the home page sets this particular button at 126px.  It just happens the text here longer than the text used anywhere else in the design.

mishaschwartz replied:
All of the buttons look different for the reasons I mentioned, not just this one:

>  there's more padding around the text, the text is vertically centred, the text is smaller

mishaschwartz commented on templates/partials/technology-card.html:
```diff
@@ -0,0 +1,32 @@
+                    <div class="d-flex flex-column flex-md-row ">
+                        {% for array_item in range(array_length) %}
+                            <div class="d-flex flex-column technology-card">
+                                <h6 class="technology-title">{{ section_title[array_item] }}</h6>
```
I think the title is bolded in figma but not here

alexandercyu replied:
The font weight and font size are the same as on Figma.  But I did remove the Helvetica Neue  font family because of the issues it caused.

mishaschwartz commented on templates/site/index.html:
```diff
@@ -1,25 +1,111 @@
     {% endwith %}
-    {% include "partials/pictures.html" %}
+
+    {% with %}
+        {% set title="Get Started"%}
```
"Getting"

alexandercyu replied:
Fixed the title

mishaschwartz reviewed (changes requested):
Ok here is the rest of my review for the main page
